apps/experiments/models.py

```diff
@@ -669,9 +677,27 @@ def create_new_version(self, version_description: str | None = None, make_defaul
     def _copy_attr_to_new_version(self, attr_name, new_version: "Experiment"):
         """Copies the attribute `attr_name` to the new version by creating a new version of the related record and
         linking that to `new_version`
+
+        If the related field's version matches the current value, link it to the new experiment version; otherwise,
+        create a new version of it.
+        Q: Why?
+        A: When a new experiment version is created, the a new version is also created for the related field. If no
+        new changes was made to this new version by the time we want to create another version of the experiment, it
+        would make sense to add the already versioned related field to the versioned experiment instead of creating yet
+        another version of it.
         """
-        if instance := getattr(self, attr_name):
-            setattr(new_version, attr_name, instance.create_new_version())
+        attr_instance = getattr(self, attr_name)
+        if not attr_instance:
+            return
+
+        latest_attr_version = attr_instance.latest_version
+
+        if latest_attr_version and not differs(
+            attr_instance, latest_attr_version, exclude_model_fields=self.DEFAULT_EXCLUDED_KEYS
```

Comment:
This should use the exclude keys from the related class (in case it is different from this class)

Reply:
https://github.com/dimagi/open-chat-studio/pull/692/commits/d6843fb0869f40ae7603b0f94091ca98ac457994